packages/mock-block-dock/README.md

```diff
@@ -0,0 +1,43 @@
+# Mock Block Dock
+
+A component which provides mocks for testing [Block Protocol](https://blockprotocol.org) blocks.
+
+## Usage
+
+When developing a block, wrap it in the embedder and pass your block its initial props:
+
+```jsx
+<MockBlockDock>
+  <TestBlock {...props} />
+</MockBlockDock>
+```
+
+The embedder will automatically pass the following Block Protocol functions to your block:
+
+- `getEntities`
+- `createEntities`
+- `deleteEntities`
+- `updateEntities`
+- `getLinks`
+- `createLinks`
+- `deleteLinks`
+- `updateLinks`
+- `uploadFile`
+
+For example, to update your block's props, get `entityId` and `updateEntities` from props and call:
+
+```typescript
+updateEntities?.([{ entityId, data: { ...newProps } }]);
```

Comment:
`?.` is the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) - it's a way of avoiding errors if the thing is null or undefined (which it might be here)

Reply:
oh I didn't realise you could optional chain on a function call like that